nix/bin/install-openocd-vexriscv-udev

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# SPDX-FileCopyrightText: 2022 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Setup plugdev group, group membership, and udev rules to get openocd-vexriscv
+# to work nicely with our JTAG debugger:
+#
+#   https://www.digikey.nl/en/products/detail/seeed-technology-co.,-ltd/114991786/10060366
+#
+set -euf -o pipefail
+
+DEV_GROUP=plugdev
+UDEV_RULES_BASENAME="60-openocd.rules"
+UDEV_RULES_D="/etc/udev/rules.d"
+
+exec_print() {
+  echo "> $@"
+  "$@"
+}
+
+set +e
+getent group "${DEV_GROUP}" >/dev/null 2>&1
+GETENT_EXIT_CODE=$?
+set -e
+
+if [ $GETENT_EXIT_CODE -eq 0 ]; then
+  echo "'${DEV_GROUP}' group already exists, doing nothing"
+else
+  exec_print sudo groupadd "${DEV_GROUP}"
+fi
+
+CURRENT_USER="$(whoami)"
+if groups "${CURRENT_USER}" | grep -q '\b'"${DEV_GROUP}"'\b'; then
+  echo "'${CURRENT_USER}' is already a member of '${DEV_GROUP}', doing nothing"
+else
+  exec_print sudo usermod -aG "${DEV_GROUP}" "${CURRENT_USER}"
+  exec_print newgrp "${DEV_GROUP}"
+fi
+
+if [ -f "${UDEV_RULES_D}/${UDEV_RULES_BASENAME}" ]; then
+  echo "'${UDEV_RULES_D}/${UDEV_RULES_BASENAME}' already exists, doing nothing"
+else
+  exec_print sudo cp "$OPENOCD_DIST/share/openocd/contrib/${UDEV_RULES_BASENAME}" "${UDEV_RULES_D}"
+  exec_print sudo udevadm control --reload-rules
+fi
```

Comment:
I'm conflicted whether this belongs in this PR.

Reply:
Discussed this offline: while this does belong in another PR, I'm feeling a bit overwhelmed by all the changes there, so I'd like to merge it separately.